controllers/druid/handler.go

```diff
@@ -1287,6 +1289,22 @@ func makeLabelsForNodeSpec(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid,
 	return labels
 }
 
+// makeAnnotationsForReplicationController returns the annotations for a Deployment or StatefulSet
+// If a given key is set in both the DruidSpec and DruidNodeSpec, the node-scoped value will take precedence.
+func makeAnnotationsForReplicationController(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid) map[string]string {
+	var annotations = map[string]string{}
+
+	if nodeSpec.ReplicationControllerAnnotations != nil && m.Spec.ReplicationControllerAnnotations != nil {
```

Comment:
Why is `nodeSpec.ReplicationControllerAnnotations != nil` condition relevant here?

Reply:
Thats a good question - I just copied this logic from over here:
https://github.com/datainfrahq/druid-operator/blob/09aeeb3c1e85a72a0ea9538e975f0a6faa3e6f95/controllers/druid/handler.go#L1270-L1281

And now that I look at it, I'm not sure why this pattern appears there either - any ideas?

Anyways - thanks for point this out, I'll get it cleaned up today.